src/main/java/io/kestra/plugin/langchain4j/embeddings/MongoDBAtlas.java

```diff
@@ -0,0 +1,132 @@
+    @Schema(title = "The host")
+    private Property<String> host;
+
+    @Schema(title = "The database")
+    private Property<String> database;
+
+    @Schema(title = "The connection string options")
+    private Property<Map<String, Object>> options;
+
+    @NotNull
+    @Schema(title = "The collection name")
+    private Property<String> collectionName;
+
+    @NotNull
+    @Schema(title = "The index name")
+    private Property<String> indexName;
+
+    @Schema(title = "The metadata field names")
+    private Property<List<String>> metadataFieldNames;
+
+    @Schema(title = "Create the index")
+    private Property<Boolean> createIndex;
+
+    @Override
+    public EmbeddingStore<TextSegment> embeddingStore(RunContext runContext, int dimension, boolean drop) throws IOException, IllegalVariableEvaluationException {
+
+        var mongoClient = MongoClients.create(buildUri(runContext));
+
+        var renderedCreateIndex = runContext.render(createIndex).as(Boolean.class).orElse(false);
+        var store = MongoDbEmbeddingStore.builder()
+            .fromClient(mongoClient)
+            .databaseName(runContext.render(database).as(String.class).orElseThrow())
+            .collectionName(runContext.render(collectionName).as(String.class).orElseThrow())
+            .indexName(runContext.render(indexName).as(String.class).orElseThrow())
+            .createIndex(renderedCreateIndex)
+            .indexMapping(
+                metadataFieldNames != null ?
+                    IndexMapping.builder()
+                        .dimension(dimension)
+                        .metadataFieldNames(new HashSet<>(runContext.render(metadataFieldNames).asList(String.class)))
+                        .build() : null
+            )
+            .build();
+
+        if (renderedCreateIndex) {
+            // Creating a vector search index can take up to a minute, so this delay allows the index to become queryable
+            try {
+                Thread.sleep(15000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        if (drop) {
+            store.removeAll();
+        }
+
+        return store;
+    }
+
+    private String buildUri(RunContext runContext) throws IllegalVariableEvaluationException {
+
+        // Format: mongodb+srv://[username:password@]host[/[database][?options]]
+
+        var username = runContext.render(this.username).as(String.class).orElse(null);
+        var password = runContext.render(this.password).as(String.class).orElse(null);
+        var host = runContext.render(this.host).as(String.class).orElseThrow();
+        var database = runContext.render(this.database).as(String.class).orElseThrow();
+        var options = runContext.render(this.options).asMap(String.class, Object.class);
+
+        boolean isAtlas = host.endsWith(".mongodb.net");
+        var scheme = isAtlas ? "mongodb+srv" : "mongodb"; // mongodb here is just for test purposes
```

Comment:
I think `mongo+svr` can be used for more than MongoDB atlas. The best would be to let the user configure the URI by itself instead of crafting it for it as there are a lot of different URI possible.

Reply:
Yes at first I thought to let the user set the full uri but it contains info I need elsewhere such as the database. And plus I applied the "Never trust user!" rule who can put a lot of error in such a connection string, what do you think?